include/ignition/gui/qml/PlottingInterface.qml

```diff
@@ -0,0 +1,250 @@
+      anchors.top: parent.top
+      anchors.margins: 10
+
+      width: 50
+      height: 50
+      radius: width/2
+      color: (Material.theme == Material.Light) ? Material.color(Material.Grey, Material.Shade700)
+                                                : Material.accentColor
+
+
+      Text {
+          text:  "+"
+          font.weight: Font.Bold
+          font.pixelSize: parent.width/2
+          color: "white"
+          anchors.centerIn: parent
+      }
+
+      MouseArea {
+          anchors.fill: parent
+          hoverEnabled: true
+          onEntered: {
+              addBtn.opacity = 0.8;
+              cursorShape = Qt.PointingHandCursor;
+          }
+          onExited: {
+              addBtn.opacity = 1;
+              cursorShape =  Qt.ArrowCursor
+          }
+
+          onClicked: {
+              if (Object.keys(charts).length == 2)
+              {
+                  multiChartsMode = true;
+                  var firstChart = true;
+                  for (var i = 0; i < chartsLayout.children.length; i++)
+                  {
+                      // skip the first one
+                      if (firstChart)
+                      {
+                          firstChart = false;
+                          // make that chart has a full size
+                          chartsLayout.heightFactor = 1
+                          continue;
+                      }
+                      var chart = chartsLayout.children[i];
+                      reallocateChart(chart);
+                  }
+              } // end if
+
+              addChart();
+          }
+      }
+  }
+
+  function addChart()
+  {
+      main.idIncrementor++;
+
+      var chartComponent = Qt.createComponent("Chart.qml");
+      var chartObject;
+      // if the mode is many charts that are organized in horizontal layout
+      if (multiChartsMode)
+      {
+          // add it to the horizontal layout
+          chartObject = chartComponent.createObject(rowChartsLayout);
+          chartObject.width = 200;
+          chartObject.height = Qt.binding( function() {return rowCharts.height * 0.9} );
+          chartObject.y = Qt.binding( function() {return (rowCharts.height - chartObject.height)/2} );
+          chartObject.multiChartsMode = true;
+          chartObject.fillPlotInOrOut()
+      }
+      else
+      {
+          // if normal mode, add the chart to the normal vertical layout
+          chartObject = chartComponent.createObject(chartsLayout);
+          chartObject.height = Qt.binding( function() {return chartsLayout.height / chartsLayout.heightFactor});
+          chartObject.width = Qt.binding( function() {return chartsLayout.width});
+          // to change the height of each vertical chart
+          chartsLayout.heightFactor ++ ;
+      }
+
+      // Chart ID
+      chartObject.chartID = main.idIncrementor;
+      charts[idIncrementor] = chartObject;
+      // Signals and Slots
+      chartObject.subscribe.connect(main.onSubscribe);
```

Comment:
Just curious, you couldn't connect directly to `PlottingIface`'s `subscribe` slot here directly? The same goes for all the other connections being wrapped.

Reply:
Yes i can do that, i do this to make the interactions between the UI and the Class through the PlottingInterface.qml only 👀